src/System.Management.Automation/engine/SessionStateProviderAPIs.cs

```diff
@@ -1510,285 +1452,6 @@ private void NewProviderEntry(ProviderInfo provider)
-            }
-            catch (ProviderNotFoundException)
-            {
-                return;
-            }
-
-            try
-            {
-                // First get an instance of the provider to make sure it exists
-                Provider.CmdletProvider providerBase = GetProviderInstance(provider);
-
-                if (providerBase == null)
-                {
-                    ProviderNotFoundException e = new ProviderNotFoundException(
-                        providerName,
-                        SessionStateCategory.CmdletProvider,
-                        "ProviderNotFound",
-                        SessionStateStrings.ProviderNotFound);
-                    context.WriteError(new ErrorRecord(e.ErrorRecord, e));
-
-                    errors = true;
-                }
-                else
-                {
-                    // See if there are any drives present for the provider
-
-                    int driveCount = 0;
-                    foreach (PSDriveInfo drive in GetDrivesForProvider(providerName))
-                    {
-                        if (drive != null)
-                        {
-                            ++driveCount;
-                            break;
-                        }
-                    }
-
-                    if (driveCount > 0)
-                    {
-                        if (force)
-                        {
-                            // Forcefully remove all the drives
-
-                            foreach (PSDriveInfo drive in GetDrivesForProvider(providerName))
-                            {
-                                if (drive != null)
-                                {
-                                    RemoveDrive(drive, true, null);
-                                }
-                            }
-                        }
-                        else
-                        {
-                            errors = true;
-
-                            // Since there are still drives associated with the provider
-                            // the provider cannot be removed
-
-                            SessionStateException e = new SessionStateException(
-                                providerName,
-                                SessionStateCategory.CmdletProvider,
-                                "RemoveDrivesBeforeRemovingProvider",
-                                SessionStateStrings.RemoveDrivesBeforeRemovingProvider,
-                                ErrorCategory.InvalidOperation);
-                            context.WriteError(new ErrorRecord(e.ErrorRecord, e));
-
-                            return;
-                        }
-                    }
-
-                    // Now tell the provider that they are going to be removed by
-                    // calling the Stop method
-
-                    try
-                    {
-                        providerBase.Stop(context);
-                    }
```

Comment:
It's hard to be certain all of this code can be safely deleted, but this line is worrisome - it potentially calls third party code to clean up resources.

Reply:
Best that I could tell, RemoveProvider() is only used for RunspaceConfiguration

Reply:
I think there is a bug in `LocalRunspace.DoCloseHelper` then - there is an early return in there, and if it always returns, there are multiple things we're not doing at shutdown, e.g. we're not firing the exiting event.

Bottom line - not hitting the code doesn't mean much w.r.t. RunspaceConfiguration, at least based on my review of the code and quick debugging of process exit.

Reply:
I'll change the DoCloseHelper() method.  Based on the comments, it's preventing stopping transcription until all the runspaces are closed, so I'll just use a flag to catch if there's any open runspaces and will continue with the rest of DoCloseHelper() to shutdown.